sdk/go/pulumi/context.go

```diff
@@ -388,7 +388,14 @@ func (ctx *Context) registerTransform(t ResourceTransform) (*pulumirpc.Callback,
 			opts.IgnoreChanges = rpcReq.Options.IgnoreChanges
 			opts.Parent = parent
 			opts.PluginDownloadURL = rpcReq.Options.PluginDownloadUrl
-			opts.Protect = rpcReq.Options.Protect
+
+			// TODO: The Go public API can't express the optionality that the wire protocol can
```

Comment:
Do we have an issue for this that we can link?

Reply:
Well https://github.com/pulumi/pulumi/issues/18833 kinda, although that was for the other cases we've already fixed up and the ones fixed in this PR. So we should probably close that one off and write a new issue for this.